cosmos/providers/databricks/notebook.py

```diff
@@ -107,7 +107,9 @@ def convert_to_databricks_workflow_task(
                 if t in relevant_upstreams
             ],
             "job_cluster_key": self.job_cluster_key,
-            "timeout_seconds": 0,
+            "timeout_seconds": int(self.execution_timeout.total_seconds())
+            if self.execution_timeout
+            else 0,
```

Comment:
Ah, this is interesting, so Databricks considers timeout=0 to be timeout disabled, not for it to instantly timeout :)

Reply:
It would be great to add a unit test for this.

Reply:
Yes, Databricks considers 0 to mean forever. That was also the previous value as you can see before my change.
The latest documentation is not that great and also there is no link to previous documentation. But, I found community answer which says so https://community.databricks.com/s/question/0D58Y00009VZwZoSAL/what-is-the-timeout-for-dbutilsnotebookrun-timeout-0-

I also tested that the job does not timeout immediately when set to 0 and succeeds until completion.